pkg/controller/apiextensions/composite/reconciler.go

```diff
@@ -366,7 +366,15 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	cds := make([]*composed.Unstructured, len(refs))
 	for i := range refs {
 		cd := composed.New(composed.FromReference(refs[i]))
-		if err := r.composed.Render(ctx, cr, cd, comp.Spec.Resources[i]); err != nil {
+
+		ct := comp.Spec.Resources[i]
+
+		if err := ct.InjectCompositionData(comp.Spec); err != nil {
+			log.Debug(errInjectCompData, "error", err)
+			r.record.Event(cr, event.Warning(reasonCompose, err))
+		}
```

Comment:
Do we want to return early from the reconcile if this injection fails? At the moment it seems like we'll proceed and attempt to do what we can with the presumably partially complete patches. If so we should add a test to cover this case.

Reply:
Yes. I think I even wrote some blurb about that in one of the tests, or one of the comments, and then forgot to actually implement it 🤦 Will Fix